### PR TITLE
Fix two remote_timeout configs in ingester_client block

### PR DESCRIPTION
**Pull request by RangerCD**
**What this PR does / why we need it**:

1. Expose `remote_timeout` under `pool_config` block in yaml config file, as expected in documentation.
2. Fix mismatched flags `-ingester.client.healthcheck-timeout` and `-ingester.client.timeout` in documentation.

**Which issue(s) this PR fixes**:

None

**Special notes for your reviewer**:

`remotetimeout` inside `pool_config` has been renamed to `remote_timeout`. This change does not affect any previous config file, since `remotetimeout` has never been actually exposed in yaml before.

loki-local-config.yaml:
```
ingester_client:
  pool_config:
    remotetimeout: 1s

auth_enabled: false
...
```
Running with latest version:
```
> loki -config.file=./loki-local-config.yaml
failed parsing config: ./loki-local-config.yaml: yaml: unmarshal errors:
  line 3: field remotetimeout not found in type distributor.PoolConfig
```

**Checklist**
- [x] Documentation added
- [ ] Tests updated
- [ ] Add an entry in the `CHANGELOG.md` about the changes.

### CODE REVIEW

**Review by kavirajk — APPROVED** [review]
LGTM :+1: Thanks @RangerCD. Now it's more clear :)

**Comment by kavirajk** [comment]
cc @KMiller-Grafana

File: docs/sources/configuration/_index.md [hunk 1]
```diff
@@ -1006,10 +1006,11 @@ pool_config:
   # How quickly a dead client will be removed after it has been detected
   # to disappear. Set this to a value to allow time for a secondary
   # health check to recover the missing client.
-  [remotetimeout: <duration>]
+  # CLI flag: -ingester.client.healthcheck-timeout
+  [remote_timeout: <duration>]
```
Comment (cyriltovena) [inline docs/sources/configuration/_index.md:1010]: Should we add the default of 1s ?

Comment (RangerCD) [reply docs/sources/configuration/_index.md]: It should be OK, `pool_config` only exists here. Added in [3c8e584](https://github.com/grafana/loki/pull/5418/commits/3c8e5841167b027ff64bda22d88bd69488935767).

**Pull request merged**